erpnext/accounts/doctype/payment_request/payment_request.py

```diff
@@ -164,53 +163,48 @@ def before_submit(self):
 		elif self.payment_request_type == "Inward":
 			self.status = "Requested"
 
-		if self.payment_request_type == "Inward":
-			if self.payment_channel == "Phone":
-				self.request_phone_payment()
-			else:
-				self.set_payment_request_url()
+		if self.payment_request_type == "Inward" and self.payment_gateway:
+			controller = _get_payment_gateway_controller(self.payment_gateway)
+			tx_data = self.get_tx_data()
+			controller.on_refdoc_submission(tx_data)  # preflight check
+			self.integration_request = controller.initiate_payment(tx_data)
+			# checkout page can also be the target of a gatway initialized flow
+			# gateways also may return None if appropriate
+			self.payment_url = controller.get_payment_url(self.integration_request)
+			if not controller.is_user_flow_initiation_delegated(self.integration_request):
 				if not (self.mute_email or self.flags.mute_email):
 					self.send_email()
 					self.make_communication_entry()
 
-	def request_phone_payment(self):
-		controller = _get_payment_gateway_controller(self.payment_gateway)
-		request_amount = self.get_request_amount()
-
-		payment_record = dict(
-			reference_doctype="Payment Request",
-			reference_docname=self.name,
-			payment_reference=self.reference_name,
-			request_amount=request_amount,
-			sender=self.email_to,
-			currency=self.currency,
-			payment_gateway=self.payment_gateway,
-		)
-
-		controller.validate_transaction_currency(self.currency)
-		controller.request_for_payment(**payment_record)
```

Comment:
The phone payment gateway must initiate the user flow via `controller.initiate_payment(tx_data)` and return `controller.is_user_flow_initiation_delegated(self.integration_request) is True` to signal to ERPNext to release the flow control to the controller early for flow initialization.